### PR TITLE
Add -q option to `docker service ps`

### LINKED ISSUE #27643: Add -q option to `docker service ps`
Right now, to get task ids, you have to do something like this:

```
docker service ps -f desired-state=running  admiring_knuth | tail +2 |cut -d ' ' -f 1
```

We need a `-q` option that removes headers and only does ID.

### PR DESCRIPTION
**Pull request by gaocegege**
**\- What I did**

fixes #27643

**\- How to verify it**

```
$ docker swarm init
$ docker service create --name redis --replicas=5 redis:3.0.6
$ docker service ps -q redis
```

**\- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: cli/command/task/print.go [hunk 1]
```diff
@@ -93,18 +114,21 @@ func Print(dockerCli *command.DockerCli, ctx context.Context, tasks []swarm.Task
...
-			indentedName,
-			task.Spec.ContainerSpec.Image,
-			nodeValue,
-			command.PrettyPrint(task.DesiredState),
-			command.PrettyPrint(task.Status.State),
-			strings.ToLower(units.HumanDuration(time.Since(task.Status.Timestamp))),
-			taskErr,
-		)
+		if quiet {
+			fmt.Fprintln(out, indentedName)
```
Comment (gaocegege) [inline cli/command/task/print.go]: >In addition, if quiet, maybe we do not have to process the err things. Just print when constructing of indentedName finishes. 😄

🤔 Yeah, there is no need to process errs.

>I think `indentedName = fmt.Sprintf(" \\_ %s", indentedName)` above may be displayed via this line of code. If that, I have no idea if it is reasonable. WDYT @gaocegege

Ah, okay I think we should pass the indent logic when `ps -q`

```
		if quiet {
			fmt.Fprintln(out, indentedName)
			continue
		}
		// Indent the name if necessary
		indentedName := name
		logrus.Debugf("DEBUG: %d, %d", prevSlot, task.Slot)
		// Since the new format of the task name is <ServiceName>.<Slot>.<taskID>, we should only compare
		// <ServiceName> and <Slot> here.
		if prevServiceName == serviceName {
			indentedName = fmt.Sprintf(" \\_ %s", indentedName)
		}
		prevServiceName = serviceName
		prevSlot = task.Slot
``` 

Then the result would be 

```diff
redis.1.mwvj0azq00lcydntsv79uj2k6
-  \_ redis.1.zk9fhepa7cgnoex3pgn0sngir
+ redis.1.zk9fhepa7cgnoex3pgn0sngir
```
```

Or do you have a better idea about indented name?

**Comment by allencloud** [comment]
>
We need a -q option that removes headers and only does ID.

I am afraid `redis.1.` in `redis.1.6u87kp12lrw1ikrejqti51g8y` is not included in task id. @gaocegege

**Comment by gaocegege** [comment]
PTAL

```
$ docker service ps redis
NAME                               IMAGE        NODE          DESIRED STATE  CURRENT STATE            ERROR
redis.1.xrlbr5skxm5mfquugv7igtw21  redis:3.0.6  56a8544282d8  Running        Preparing 9 seconds ago
redis.2.69bi767slr5fmpb37n78zop4m  redis:3.0.6  56a8544282d8  Running        Preparing 9 seconds ago
$ docker service ps -q redis
xrlbr5skxm5mfquugv7igtw21
69bi767slr5fmpb37n78zop4m
```

**Comment by LK4D4** [comment]
LGTM
I'm not sure what to do with docs, though. ping @thaJeztah

**Comment by thaJeztah** [comment]
> I'm not sure what to do with docs, though. 

This would only need changes to the reference docs (man-page for this is generated); @gaocegege can you update https://github.com/docker/docker/blob/master/docs/reference/commandline/service_ps.md ?


It also needs a change to the bash and zsh completion scripts, so /cc @albers @sdurrheimer

**Comment by gaocegege** [comment]
re @thaJeztah 

I have updated the docs. And I found there is no `-a` option now, so I removed it and add `-q`.

[preview](https://github.com/gaocegege/docker/blob/636d2bfb2d46eae552746c8b973e94a0a8a08525/docs/reference/commandline/service_ps.md)

**Comment by sdurrheimer** [comment]
zsh completion LGTM

File: contrib/completion/bash/docker [hunk 2]
```diff
@@ -2462,7 +2462,7 @@ _docker_service_ps() {
 
 	case "$cur" in
 		-*)
-			COMPREPLY=( $( compgen -W "--all -a --filter -f --help --no-resolve --no-trunc" -- "$cur" ) )
+			COMPREPLY=( $( compgen -W "--quiet -q --filter -f --help --no-resolve --no-trunc" -- "$cur" ) )
```
Comment (albers) [inline contrib/completion/bash/docker]: Please preserve alphabetical order here.

File: contrib/completion/zsh/_docker [hunk 3]
```diff
@@ -1167,7 +1167,7 @@ __docker_service_subcommand() {
         (ps)
             _arguments $(__docker_arguments) \
                 $opts_help \
-                "($help -a --all)"{-a,--all}"[Display all tasks]" \
+                "($help -q --quiet)"{-q,--quiet}"[Only display IDs]" \
```
Comment (gaocegege) [inline contrib/completion/zsh/_docker]: OK

**Review by albers — APPROVED** [review]
bash completion LGTM, Thanks!

**Review by sdurrheimer — APPROVED** [review]
re-LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 👼
/cc @aaronlehmann @stevvooe

File: cli/command/task/print.go [hunk 4]
```diff
@@ -50,6 +52,27 @@ func Print(dockerCli *command.DockerCli, ctx context.Context, tasks []swarm.Task
...
 
+	if err := print(writer, ctx, tasks, resolver, noTrunc); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// PrintQuiet shows task list in a quiet way.
+func PrintQuiet(dockerCli *command.DockerCli, tasks []swarm.Task) error {
```
Comment (stevvooe) [inline cli/command/task/print.go]: No need to export this.

**Review by stevvooe — CHANGES_REQUESTED** [review]
Please don't export functions that don't need to be exported.

File: cli/command/task/print.go (hunk 4 above)
Comment (stevvooe) [inline cli/command/task/print.go]: I see: this package is useless.

Just move `print.go` to the service package. This isn't used elsewhere.

Comment (stevvooe) [inline cli/command/task/print.go]: No reason to preserve the package import. These functions are a for loop and println.

Perhaps, this is outside the scope of this pr. Let's leave as is and fix in another pr.

**Review by stevvooe — APPROVED** [review]

**Comment by stevvooe** [comment]
LGTM

If someone wants de-leftpad the task package, that would be awesome!

**Comment by vdemeester** [comment]
Moving this to docs-review then 👼

File: cli/command/service/ps.go [hunk 5]
```diff
@@ -32,6 +33,7 @@ func newPsCommand(dockerCli *command.DockerCli) *cobra.Command {
 		},
 	}
 	flags := cmd.Flags()
+	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "Only display IDs")
```
Comment (thaJeztah) [inline cli/command/service/ps.go]: Looks good, wondering if we should use `Only display task IDs` (there's so many ID's related to this overview, it became a bit confusing to me)

@vdemeester wdyt?

Comment (vdemeester) [inline cli/command/service/ps.go]: ah right, I do think `Only display task IDs` would make it a little bit clearer indeed 👼

Comment (gaocegege) [inline cli/command/service/ps.go]: Ah, OK

I would add it.

**Comment by cpuguy83** [comment]
Can you squash commits, please?

**Comment by thaJeztah** [comment]
LGTM after commits are squashed

**Review by vdemeester — APPROVED** [review]
LGTM 🐸
